doc/tutorial/single_cell_bluepyopt.rst

```diff
@@ -0,0 +1,288 @@
+   :dedent:
+   :lines: 20
+
+Axon replacement
+----------------
+
+A popular modeling choice for cell optimization with the Neuron simulator in BluePyOpt is to substitute the axon by a surrogate morphology that is computationally less expensive. The corresponding composite morphology can be obtained in Arbor by substituting the axon of the original morphology with the exported replacement using the segment tree editing primitives :func:`arbor.segment_tree.split_at` and :func:`arbor.segment_tree.join_at`. The axon-replacement morphology is automatically exported to ACC when calling ``create_acc`` on a cell model with a morphology that has ``do_replace_axon`` set to ``True``. If the axon-replacement exported by BluePyOpt is not satisfactory, alternatively, the user can also define his/her own morphology in ACC to use as a replacement.
+
+To perform axon replacement, we first have to find the (unique) root segment of this region in the morphology. We can then split the segment tree at the axon root into two separate segment trees - a pruned one and one for the original axon.
+
+.. code-block:: python
+
+   # get unique axon root (tag for axon is 2)
+   axon_root = morpho.segment_tree.tag_roots(2)[0]
+
+   # prune morphology at axon root
+   pruned_st, axon_st = morpho.segment_tree.split_at(axon_root)
+
+Then we can determine the parent of the axon root to join the replacement onto, which we load from ACC. Alternatively, the user can build/supply a custom replacement here.
+
+.. code-block:: python
+
+   # parent segment to join onto
+   axon_parent = morpho.segment_tree.parents[axon_root]
+
+   # load replacement morphology
+   axon_replacement = arbor.load_component(
+      cell_json['morphology']['replace_axon']).component
+   axon_replacement_st = replacement.to_segment_tree()
+
+   # pruned segment tree with axon replacement
+   joined_st = pruned_st.join_at(
+      axon_parent, axon_replacement_st)
+
+   # final morphology
+   morpho_axon_replaced = arbor.morphology(joined_st)
+
+The steps presented here are included when loading an exported model with the above ``read_acc`` function from BluePyOpt.
+
+
+Define locations in the GUI
+---------------------------
+
+BluePyOpt makes use of the Neuron simulator's features to define custom locations for stimuli and voltage probes that can be challenging to translate to Arbor. As an example, the layer-5 pyramidal cell optimization problem has multiple voltage probes located distally on the apical dendrite at a fixed distance from the soma as described in `config/protocols.json <https://github.com/BlueBrain/BluePyOpt/blob/master/examples/l5pc/config/protocols.json>`_.
+
+To describe these locations accurately in Arbor, we will use the Arbor GUI. For this purpose, we create an :class:`arbor.cable_cell` and export it to ACC so that it can be loaded in the GUI,
+
+.. code-block:: python
+
+   cell = arbor.cable_cell(morpho, labels, decor)
+   arbor.write_component(cell, 'path/to/l5pc_cable_cell.acc')
```

Comment:
Bit surprising that `cable_cell`, `label_dict` and `decor` are `component`s, but `morphology` isn't.

Reply:
Morph not being one is made into a specific point a bit up (`load_asc`). It's logical when you know what's happening, but not if you don't :)